src/lib/viewers/controls/page/__tests__/PageControlsForm-test.tsx

```diff
@@ -0,0 +1,110 @@
+
+            getFormButton(wrapper).simulate('click');
+            getFormInput(wrapper).simulate('change', { target: { value: newPageNumber } });
+            getFormInput(wrapper).simulate('blur');
+
+            expect(onPageSubmit.mock.calls.length).toBe(onPageSubmitCallCount);
+        });
+
+        test.each`
+            newPageNumber | onPageSubmitCallCount
+            ${''}         | ${0}
+            ${2}          | ${1}
+        `('should handle when Enter key is pressed on input', ({ onPageSubmitCallCount, newPageNumber }) => {
+            const key = ENTER;
+            const onPageSubmit = jest.fn();
+            const pageCount = 3;
+            const pageNumber = 1;
+            const preventDefault = jest.fn();
+            const stopPropagation = jest.fn();
+            const wrapper = getWrapper({ onPageSubmit, pageCount, pageNumber });
+
+            getFormButton(wrapper).simulate('click');
+            expect(getFormButton(wrapper).exists()).toBe(false);
+            expect(getFormInput(wrapper).exists()).toBe(true);
+            getFormInput(wrapper).simulate('change', {
+                preventDefault,
+                stopPropagation,
+                target: { value: newPageNumber },
+            });
+            getFormInput(wrapper).simulate('keydown', {
+                preventDefault,
+                stopPropagation,
+                key,
+            });
+
+            expect(preventDefault).toHaveBeenCalled();
+            expect(stopPropagation).toHaveBeenCalled();
+            expect(onPageSubmit.mock.calls.length).toBe(onPageSubmitCallCount);
+            expect(getFormButton(wrapper).exists()).toBe(true);
+            expect(getFormInput(wrapper).exists()).toBe(false);
+        });
+
+        test('should handle Escape key when pressed on input', () => {
+            const preventDefault = jest.fn();
+            const stopPropagation = jest.fn();
+            const wrapper = getWrapper();
+            expect(getFormButton(wrapper).exists()).toBe(true);
+            expect(getFormInput(wrapper).exists()).toBe(false);
+
+            getFormButton(wrapper).simulate('click');
+            expect(getFormButton(wrapper).exists()).toBe(false);
+            expect(getFormInput(wrapper).exists()).toBe(true);
+            getFormInput(wrapper).simulate('keydown', {
+                preventDefault,
+                stopPropagation,
+                key: ESCAPE,
+            });
+
+            expect(preventDefault).toHaveBeenCalled();
+            expect(stopPropagation).toHaveBeenCalled();
+            expect(getFormButton(wrapper).exists()).toBe(true);
+            expect(getFormInput(wrapper).exists()).toBe(false);
+        });
+    });
+
+    describe('render', () => {
+        test('should render button with correct page number', () => {
+            const pageCount = 3;
+            const pageNumber = 2;
+
+            const wrapper = getWrapper({ pageCount: 3, pageNumber: 2 });
```

Comment:
The variables above don't seem very useful in this case.

Reply:
Oops, fixed!